src/js/components/homepage/visualizations/treemap/BudgetFunctions/BudgetFunctions.jsx

```diff
@@ -0,0 +1,247 @@
+            }
+            if (height < 40 || width < 60) {
+                percentView = 'none';
+            }
+
+            // Finally, create the cell
+            if (n.value !== 0) {
+                cell = (<BudgetFunctionCell
+                    {...treeProps}
+                    label={n.data.name}
+                    value={n.value}
+                    x0={n.x0}
+                    x1={n.x1}
+                    y0={n.y0}
+                    y1={n.y1}
+                    total={n.parent.value}
+                    key={i}
+                    functionID={n.data.id}
+                    color={cellColor}
+                    strokeColor={strokeColor}
+                    strokeOpacity={strokeOpacity}
+                    tooltipStyles={treeProps.tooltipStyles}
+                    toggleTooltipIn={this.toggleTooltipIn}
+                    toggleTooltipOut={this.toggleTooltipOut}
+                    textColor={textColor}
+                    textShadow={textShadow}
+                    textClass={textClass}
+                    labelView={labelView}
+                    width={width}
+                    height={height}
+                    percentView={percentView}
+                    clickable />);
+            }
+
+            return cell;
+        });
+
+        this.setState({
+            finalNodes: nodes
+        });
+    }
+
+    toggleTooltipIn(functionID) {
+        this.setState({
+            showOverlay: false,
+            hoveredFunction: functionID
+        }, () => {
+            this.buildTree(this.props);
+        });
+    }
+
+    toggleTooltipOut() {
+        this.setState({
+            showOverlay: true,
+            hoveredFunction: -1
+        }, () => {
+            this.buildTree(this.props);
+        });
+    }
+
+    toggleSubfunction(selection) {
+        this.props.toggleSubfunction(selection);
+    }
+
+    createTooltip() {
+        let tooltip = null;
+
+        if (this.state.hoveredFunction > -1) {
+            const category = _.find(
+                this.props.categories.children,
+                { id: this.state.hoveredFunction });
+            const description = _.find(
+                this.props.descriptions,
+                { id: this.state.hoveredFunction });
+            const node = _.find(
+                this.state.finalNodes,
+                { key: `${this.state.hoveredFunction}` });
+
+            tooltip = (<TreeMapTooltip
+                name={category.name}
+                value={MoneyFormatter.formatTreemapValues(category.value)}
+                percentage={MoneyFormatter.calculateTreemapPercentage(
+                    category.value, this.props.totalNumber)
+                }
+                description={description.value}
+                x={node.props.x0}
+                y={node.props.y0}
+                width={node.props.width}
+                height={(node.props.height / 2) + 50} />);
+        }
+
+        return tooltip;
```

Comment:
An alternative is to have to component always rendered, but use CSS to hide the tooltip (via state) when not needed. This also makes debugging easier since we can just force-show the tooltip when making style fixes.